examples/pipeline/rag.py

```diff
@@ -111,6 +112,6 @@ async def run(self, prompt: str) -> StringDataModel:
     result = asyncio.run(
         pipe.run({"retrieve": {"query": query}, "augment": {"query": query}})
     )
-    print(result["generate"]["result"])
+    print(result.result["generate"]["result"])
```

Comment:
Where does generate and come from here again? IMO this syntax can get confusing. Perhaps we can make it clearer with better naming? e.g.

```python
print(pipe_output.result["generate"]["result"])
```

Reply:
`generate` is the name of the component (at the moment, we include in the returned result all the output from the leaf components) and `result` is a field from the output model from that component.

Reply:
Dumb question: is there a reason why `generate` and `result` need to have a dictionary data structure here? I guess the component name could change, so `generate` makes sense. I'm wondering whether this could be more accessible

```python
print(pipe_output.result["generate"].result)
```

Reply:
This is because we store the results as dicts in the "stores" and hence we loose the data type. We could probably change that for the results that are returned by the pipeline (different from the component' "intermediate" results), but that would be another PR ;)

Reply:
Sounds good, happy to defer that